packages/labs/analyzer/src/test/javascript/classes_test.ts

```diff
@@ -0,0 +1,282 @@
+    assert.equal(member.parameters?.[0].default, undefined);
+    assert.equal(member.parameters?.[0].rest, false);
+    assert.equal(member.parameters?.[1].name, 'b');
+    assert.equal(
+      member.parameters?.[1].description,
+      'Param b description\nwith wraparound'
+    );
+    assert.equal(member.parameters?.[1].type?.text, 'boolean');
+    assert.equal(member.parameters?.[1].optional, true);
+    assert.equal(member.parameters?.[1].default, 'false');
+    assert.equal(member.parameters?.[1].rest, false);
+    assert.equal(member.parameters?.[2].name, 'c');
+    assert.equal(member.parameters?.[2].description, 'Param c description');
+    assert.equal(member.parameters?.[2].summary, undefined);
+    assert.equal(member.parameters?.[2].type?.text, 'number[]');
+    assert.equal(member.parameters?.[2].optional, false);
+    assert.equal(member.parameters?.[2].default, undefined);
+    assert.equal(member.parameters?.[2].rest, true);
+    assert.equal(member.return?.type?.text, 'string');
+    assert.equal(member.return?.description, 'Method 2 return description');
+    assert.equal(member.deprecated, 'Method 2 deprecated');
+  });
+
+  test('static method1', ({getModule}) => {
+    const dec = getModule('classes').getDeclaration('Class1');
+    assert.ok(dec.isClassDeclaration());
+    const member = dec.getStaticMethod('method1');
+    assert.ok(member?.isClassMethod());
+    assert.equal(member.summary, `Static method 1 summary`);
+    assert.equal(member.description, `Static method 1 description`);
+    assert.equal(member.parameters?.length, 3);
+    assert.equal(member.parameters?.[0].name, 'a');
+    assert.equal(member.parameters?.[0].description, 'Param a description');
+    assert.equal(member.parameters?.[0].summary, undefined);
+    assert.equal(member.parameters?.[0].type?.text, 'string');
+    assert.equal(member.parameters?.[0].default, undefined);
+    assert.equal(member.parameters?.[0].rest, false);
+    assert.equal(member.parameters?.[1].name, 'b');
+    assert.equal(member.parameters?.[1].description, 'Param b description');
+    assert.equal(member.parameters?.[1].type?.text, 'boolean');
+    assert.equal(member.parameters?.[1].optional, true);
+    assert.equal(member.parameters?.[1].default, 'false');
+    assert.equal(member.parameters?.[1].rest, false);
+    assert.equal(member.parameters?.[2].name, 'c');
+    assert.equal(member.parameters?.[2].description, 'Param c description');
+    assert.equal(member.parameters?.[2].summary, undefined);
+    assert.equal(member.parameters?.[2].type?.text, 'number[]');
+    assert.equal(member.parameters?.[2].optional, false);
+    assert.equal(member.parameters?.[2].default, undefined);
+    assert.equal(member.parameters?.[2].rest, true);
+    assert.equal(member.return?.type?.text, 'string');
+    assert.equal(
+      member.return?.description,
+      'Static method 1 return description'
+    );
+    assert.equal(member.deprecated, 'Static method 1 deprecated');
+  });
```

Comment:
All of these tests are moved from `lit-element/jsdoc_test.js` (and their duplicates that were in `vanilla-element/jsdoc_test.js` into here, so that this file is now all of the class-related tests.

The new tests are the const tests below, plus the `superClass` test, so that we have a superClass test in the basic class tests (there another in the LitElement tests, so there was some coverage, but this makes sure it works for vanilla classes also).